pkg/cmd/template/data_values_flags.go

```diff
@@ -274,26 +276,36 @@ func (s *DataValuesFlags) kvFile(kv string) (*datavalues.Envelope, error) {
 	return datavalues.NewEnvelopeWithLibRef(overlay, libRef)
 }
 
+// libraryRefAndKey separates a library reference and a key and validates that no libraryKeySep exist in the key.
+// libraryKeySep is disallowed in data value flag keys.
 func (DataValuesFlags) libraryRefAndKey(key string) (string, string, error) {
-	const (
-		libraryKeySep = ":"
-	)
-
-	keyPieces := strings.Split(key, libraryKeySep)
-
-	switch len(keyPieces) {
-	case 1:
-		return "", key, nil
+	libRef, key, err := DataValuesFlags{}.libraryRefAndString(key)
+	if err != nil {
+		return "", "", err
+	}
+	if len(strings.Split(key, libraryKeySep)) > 1 {
+		// error on a common syntax mistake
+		return "", "", fmt.Errorf("Expected at most one library-key separator '%s' in '%s'", libraryKeySep, key)
+	}
+	return libRef, key, nil
+}
 
-	case 2:
-		if len(keyPieces[0]) == 0 {
-			return "", "", fmt.Errorf("Expected library ref to not be empty")
+// libraryRefAndString separates a library reference and a string.
+// A library reference starts with ref.LibrarySep, and ends with the first occurrence of libraryKeySep.
+func (DataValuesFlags) libraryRefAndString(str string) (string, string, error) {
```

Comment:
```suggestion
func (DataValuesFlags) libraryRefAndRemainder(arg string) (string, string, error) {
```

Part one: the method name....

Trying to think of a more concrete name for the second return value... (burned through a timebox over here)

Ideally, it would be a domain term ("library ref" is a domain term; what the rest of this argument called?):
- in the one case, it's the key/value... what is that?
  - Setting
  - Assignment
  - er... KeyValuePair
- in the another case, it's the path to a file that contains key/values (`--data-values-file`)
  - path
  - (sortof) KeyValuePairs

But nothin' gellin'... so perhaps the name of the role of the value in calculations:
- Remainder
- TheRest :)

I test these ideas out by looking at the _calling_ spot to see if as a reader, I have a chance of not having to read the body of the called function to get the gist of what's going on.


Part two: the parameter name...
It's more than the key, it's the whole argument. So... `arg`?

Reply:
`libraryRefAndRemainder` is more clear 👍 , making that change also made it easier to update the godoc to be more descriptive.